extensions/coroutines-extensions/src/commonMain/kotlin/app/cash/sqldelight/coroutines/FlowExtensions.kt

```diff
@@ -58,7 +61,7 @@ fun <T : Any> Flow<Query<T>>.mapToOne(
   context: CoroutineContext = Dispatchers.Default
 ): Flow<T> = map {
   withContext(context) {
-    it.executeAsOne()
+    it.awaitAsOne()
   }
```

Comment:
Do we need the `withContext` here anymore since `awaitAsOne` suspends?

Reply:
I think so, that way the caller can still control what context the `await*` methods are executed in.

Reply:
Forgive my ignorance. Is that not something `flowOn` could solve?